### PR TITLE
authorize: additional header evaluator tests

### PR DESCRIPTION
**Pull request by calebdoxsey**
## Summary
In preparation for #5362 add some additional unit tests and a benchmark.

## Related issues
- #5357 
 

## Checklist

- [x] reference any related issues
- [x] updated unit tests
- [x] add appropriate label (`enhancement`, `bug`, `breaking`, `dependencies`, `ci`)
- [x] ready for review

### CODE REVIEW

**Comment by calebdoxsey** [comment]
On my machine this is the benchmark:

```
goos: linux
goarch: amd64
pkg: github.com/pomerium/pomerium/authorize/evaluator
cpu: AMD Ryzen 7 5800H with Radeon Graphics         
BenchmarkHeadersEvaluator-16    	    1041	   1135602 ns/op	  635476 B/op	   13231 allocs/op
PASS
ok  	github.com/pomerium/pomerium/authorize/evaluator	1.339s
```

File: authorize/evaluator/headers_evaluator_test.go [hunk 1]
```diff
@@ -20,14 +20,74 @@ import (
...
+	ctx = storage.WithQuerier(ctx, storage.NewStaticQuerier([]proto.Message{
+		&session.Session{Id: "s1", ImpersonateSessionId: proto.String("s2"), UserId: "u1"},
+		&session.Session{Id: "s2", UserId: "u2", Claims: map[string]*structpb.ListValue{
+			"name": {Values: []*structpb.Value{
+				structpb.NewStringValue("n1"),
+			}},
+		}, IssuedAt: timestamppb.New(iat)},
+		&user.User{Id: "u2", Name: "USER#2"},
+		newDirectoryUserRecord(directory.User{ID: "u2", GroupIDs: []string{"g1", "g2", "g3", "g4"}}),
+		newDirectoryGroupRecord(directory.Group{ID: "g1", Name: "GROUP1"}),
```
Comment (wasaga) [inline authorize/evaluator/headers_evaluator_test.go:53]: I think we also populate group emails 
https://github.com/pomerium/pomerium/blob/9f606a9fe92a4f12df793bd35d8679f4a4a5d0c3/authorize/evaluator/opa/policy/headers.rego#L68

**Review by wasaga — APPROVED** [review]

File: authorize/evaluator/headers_evaluator_test.go (hunk 1 above)
Comment (calebdoxsey) [reply authorize/evaluator/headers_evaluator_test.go:53]: Strangely that code appears unused. I think we did populate emails at one time but somewhere along the way that support was dropped. Probably on accident.

Comment (calebdoxsey) [reply authorize/evaluator/headers_evaluator_test.go:53]: I added some groups to confirm the existing behavior in the JWT test. Emails weren't included.

**Pull request merged**